dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py

```diff
@@ -0,0 +1,276 @@
+        endpoint = self.url + "/api/v1/entity/services"
+        services = self.get_json_response(endpoint)
+        return services
+
+    def process_component(self, response, component_type):
+        """
+        Process each component type and map those with specific data
+        :param response: Response of each component type endpoint
+        :param component_type: Component type
+        :return: create the component on stackstate API
+        """
+        urn = ''
+        if "error" not in response:
+            for item in response:
+                # special case for host type as we get some float values
+                if component_type == "host":
+                    for key in item.keys():
+                        if type(item[key]) is float:
+                            item[key] = str(item[key])
+                identifiers = []
+                data = dict()
+                externalId = item.get("entityId")
+                if component_type == "service":
+                    urn = "urn:service:/{}".format(externalId)
+                elif component_type == "process-group":
+                    urn = "urn:process-group:/{}".format(externalId)
+                elif component_type == "application":
+                    urn = "urn:application:/{}".format(externalId)
+                elif component_type == "process":
+                    urn = "urn:process:/{}".format(externalId)
+                elif component_type == "host":
+                    displayName = item.get("displayName")
+                    urn = "urn:host:/{}".format(displayName)
+                identifiers.append(urn)
+                # derive useful labels and get labels from dynatrace tags
+                labels = self.get_labels(item)
+                labels += self.get_labels_from_dynatrace_tags(item)
+                if "tags" in item:
+                    del item["tags"]
+                # prefix the labels with `dynatrace-` for all labels
+                labels = ["dynatrace-{}".format(label) for label in labels]
+                data.update(item)
+                data.update({
+                    "identifiers": identifiers,
+                    "tags": self.tags,
+                    "domain": self.domain,
+                    "environment": self.environment,
+                    "instance": self.url,
+                    "labels": labels
+                })
+                data = self.filter_data(data)
+                self.component(externalId, component_type, data)
+                self.collect_relations(item, externalId)
+        else:
+            raise DynatraceError(response["error"].get("message"), response["error"].get("code"), component_type)
+
+    def get_labels_from_dynatrace_tags(self, item):
+        """
+        Process each tag as a label in component
+        :param item: the component item to read from
+        :return: list of added tags as labels
+        """
+        tags = []
+        for tag in item.get("tags", []):
```

Comment:
Is `tags` a dictionary? so we should default to `{}` then

Reply:
no it's a list.